extension/src/help/openWebHelp.ts

```diff
@@ -0,0 +1,212 @@
+	if (selected === "") {
+		await vscode.commands.executeCommand('editor.action.addSelectionToNextFindMatch');
+		selected = editor.document.getText(editor.selection);
+	}
+	let urlPath: string = webHelpContainer.getWebHelpUrlBySymbolName(selected);
+	vscode.env.openExternal(vscode.Uri.parse(urlPath));
+}
+
+
+// This container object represents all the normalized data extracted from help.autodesk.com/view/OARX/
+export class WebHelpLibrary{
+	objects: Dictionary<WebHelpObject> = {};
+	functions: Dictionary<WebHelpFunction> = {};
+	ambiguousFunctions: Dictionary<WebHelpFunction[]> = {};
+	enumerators: Dictionary<WebHelpEntity> = {};	
+	
+	// consumes a JSON converted object into the WebHelpLibrary
+	load(obj: object): void{		
+		Object.keys(obj["objects"]).forEach(key => {
+			let newObj = new WebHelpObject(obj["objects"][key]);
+			this.objects[key] = newObj;
+		});
+		Object.keys(obj["functions"]).forEach(key => {
+			let newObj = new WebHelpFunction(obj["functions"][key]);
+			this.functions[key] = newObj;
+		});
+		Object.keys(obj["ambiguousFunctions"]).forEach(key => {
+			let newLst = [];
+			obj["ambiguousFunctions"][key].forEach(element => {
+				newLst.push(new WebHelpFunction(element));
+			});
+			this.ambiguousFunctions[key] = newLst;
+		});
+		Object.keys(obj["enumerators"]).forEach(key => {
+			let newObj = new WebHelpEntity(obj["enumerators"][key]);
+			this.enumerators[key] = newObj;
+		});
+	}
+
+	getLanguageUrlDomain(): string {
+		switch (vscode.env.language.toLowerCase()) {			
+			case "de": return "/DEU/";
+			case "es": return "/ESP/";
+			case "fr": return "/FRA/";
+			case "hu": return "/HUN/";
+			case "ja": return "/JPN/";
+			case "en": return "/ENU/";
+			case "ko": return "/KOR/";
+			case "pt": return "/PTB/";
+			case "br": return "/PTB/";
+			case "pt-br": return "/PTB/";
+			case "ru": return "/RUS/";
+			case "zh": return "/CHS/";
+			case "chs": return "/CHS/";
+			case "cn": return "/CHS/";
+			case "zh-cn": return "/CHS/";
+			case "zh-hans": return "/CHS/";
+			case "cht": return "/CHT/";
+			case "tw": return "/CHT/";
+			case "zh-tw": return "/CHT/";
+			case "zh-hant": return "/CHT/";
+			default: return "/ENU/";
+		}
+	}
+
+	getDefaultHelpLink(languageDomain: string): string{
+		let year: number = new Date().getFullYear() + 1;
+		return "https://help.autodesk.com/view/OARX/" + year.toString() + languageDomain;
+	}
+
+	// Searches the library dictionaries for a reference to the provided symbol name. If found, yields help URL relevant to that symbol, but otherwise outputs generalized help URL
+	getWebHelpUrlBySymbolName(item: string): string {
+		let symbolProfile: string = item.toLowerCase();
+		let lang: string = this.getLanguageUrlDomain();
+		if (symbolProfile in this.functions){        
+			return this.functions[symbolProfile].getHelpLink(lang);
+		} else if (symbolProfile in this.enumerators){
+			return this.enumerators[symbolProfile].getHelpLink(lang);
```

Comment:
This is a fine example of inconsistencies in the Help documentation causing 1-off scenarios that requires special rules. My goal was to provide help documentation links for the many named enumerators that really only exist whenever a method/property has referenced them. Seeing this I probably should have made them be a minimum of 5 characters or something, but in general this appears to be working exactly as intended because of the way they did the layout of the page.

![image](https://user-images.githubusercontent.com/51800232/93811240-a9610d00-fc04-11ea-914b-96c504e5b246.png)